lib/zxcvbn/matchers/l33t.rb

```diff
@@ -31,8 +31,8 @@ def matches(password)
               token = password[match.i..match.j]
               next if token.downcase == match.matched_word.downcase
               match_substitutions = {}
-              substitution.each do |substitution, letter|
-                match_substitutions[substitution] = letter if token.include?(substitution)
+              substitution.each do |s, letter|
```

Comment:
I was thinking `inject`/`collect` might be a better fit here but looking at it again; I'm not sure if it is more readable as is.

Reply:
I wondered about refactoring this method entirely but decided against it for now.